pg_ivm.c

```diff
@@ -320,3 +322,30 @@ PgIvmImmvPrimaryKeyIndexId(void)
 	return pg_ivm_immv_pkey_id;
 }
 
+/*
+ * Return the SELECT part of a IMMV
+ */
+Datum
+get_immv_def(PG_FUNCTION_ARGS)
+{
+	Oid	matviewOid = PG_GETARG_OID(0);
+	Relation matviewRel = NULL;
+	Query *query = NULL;
+	char *querystring = NULL;
+
+	matviewRel = table_open(matviewOid, AccessShareLock);
+	/* Make sure IMMV is a table. */
+	Assert(matviewRel->rd_rel->relkind == RELKIND_RELATION);
```

Comment:
Due to this assert, passing a relation other than table causes an error.
```
postgres=# \d v
                  View "public.v"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 i      | integer |           |          | 

postgres=# select get_immv_def('v');
TRAP: FailedAssertion("matviewRel->rd_rel->relkind == RELKIND_RELATION", File: "pg_ivm.c", Line: 338, PID: 37560)
```

How about just return NULL in this case, as pg_get_viewdef does?

Reply:
OK I change to NULL